addon/components/nypr-account-forms/verify.js

```diff
@@ -0,0 +1,46 @@
+import Component from 'ember-component';
+import computed from 'ember-computed';
+import get from 'ember-metal/get';
+import fetch from 'fetch';
+import { rejectUnsuccessfulResponses } from 'nypr-account-settings/utils/fetch-utils';
+import layout from '../../templates/components/nypr-account-forms/verify';
+import messages from 'nypr-account-settings/validations/nypr-accounts/custom-messages';
+
+export default Component.extend({
+  layout,
+  session: null,
+  verificationCode: null,
+  emailId: null,
+  authAPI: null,
+  onFailure: () => {},
+  onSuccess: () => {},
+  APIUrl: computed('authAPI', 'email_id', function() {
+    return `${get(this, 'authAPI')}/membership/email/${get(this, 'emailId')}/verify`;
+  }),
+  didReceiveAttrs() {
+    this.verifyEmail(get(this, 'APIUrl'), get(this, 'verificationCode'));
+  },
+  verifyEmail(url, verificationCode) {
+    this.get('session').authorize('authorizer:nypr', (header, value) => {
```

Comment:
the authorize block will be run once for every authorizer, so you shouldn't actually make your api call in here. use it to construct your headers object.